frontends/p4/unusedDeclarations.h

```diff
@@ -90,29 +91,35 @@ class CollectUsedDeclarations : public Inspector, ResolutionContext {
  * Additionally, IR::Declaration_Instance nodes for extern instances are not
  * removed but still trigger warnings.
  *
- * If @warned is non-null, unused IR::P4Table and IR::Declaration_Instance
- * nodes are stored in @warned if they are unused and removed by this pass.  A
- * compilation warning is emitted when a new node is added to @warned,
+ * A compilation warning is emitted when a new node is added to @warned,
  * preventing duplicate warnings per node.
  *
  * @pre Requires an up-to-date ReferenceMap.
  */
-class RemoveUnusedDeclarations : public Transform, ResolutionContext {
+class UnusedDeclarations : public Transform, ResolutionContext {
  protected:
     const UsedDeclSet &used;
 
-    /** If not null, logs the following unused elements in @warned:
+    /// If true, removes unused instances.
+    bool removeUnused;
+
+    /// If true, emits warnings about unused instances.
+    bool warnUnused;
+
+    /// If true, emits warnings about removed instances.
+    bool warnRemoved;
```

Comment:
Maybe should be called `infoRemoved` as it actually outputs info messages for removed instances, not warnings.  Should at least fix the comment.